### PR TITLE
Fix concatenation of raws with np.nan in the dev-head-transform

### PR DESCRIPTION
**Pull request by mscheltienne**
Reported here: https://mne.discourse.group/t/concatenate-raws-reported-raws-1-info-dev-head-t-differs/7838

### CODE REVIEW

**Comment by mscheltienne** [comment]
Failures seems unrelated.

File: mne/_fiff/meas_info.py [hunk 1]
```diff
@@ -3729,7 +3729,10 @@ def _ensure_infos_match(info1, info2, name, *, on_mismatch="raise"):
     if (info1["dev_head_t"] is None) != (info2["dev_head_t"] is None) or (
```
Comment (drammock) [inline mne/_fiff/meas_info.py:3729]: passing comment / nitpick: to me xor infix (`^`) seems clearer than `!=` here

```suggestion
    if (info1["dev_head_t"] is None) ^ (info2["dev_head_t"] is None) or (
```

**Review by drammock — APPROVED** [review]
code changes look good, thanks!

**Comment by larsoner** [comment]
Unrelated failures should be fixed in #12202

**Review by wmvanvliet — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @mscheltienne !

**Pull request merged**